### PR TITLE
project: saving history docs

### PR DESCRIPTION
**Pull request by GiovannaMonti**
## Description

This PR adds a section to describe the new Saving History feature for the Enhanced Project Workflow

## Pull Request Type

- [x] Documentation content changes
- [ ] Bugfix / Missing Redirects
- [ ] Docusaurus site code changes

## PR Checklist

- [x] The commit message follows our guidelines included in the [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md#how-to-submit-a-pr)
- [x] All tests of Lint, cspell and check-content pass. ([How to launch them?](../blob/main/CONTRIBUTING.md#content-checks))
- [x] No sensitive content has been committed

### CODE REVIEW

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md [hunk 1]
```diff
@@ -87,6 +81,14 @@ Versions can only be deleted by Project Administrators.
 
 ![Versions management page](img/versions-overview.png)
 
+### Revision history
+
+To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. To keep track of how a revision has changed over time, the user can access its [revision history](#revision-history), which contains past snapshots of the Project configuration.
```
Comment (GiovannaMonti) [inline docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:86]: I don't think it is necessary to link the #revision-history section here, since we are already inside it

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md [hunk 2]
```diff
@@ -87,6 +81,14 @@ Versions can only be deleted by Project Administrators.
...
+
+![Revision History modal](img/revision-history.png)
```
Comment (GiovannaMonti) [inline docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:88]: this screenshot shows a little bug on the last column, where buttons should be aligned to the right. Let's try to take a snapshot without the bug (it should suffice to reduce the browser width a little bit)

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md [hunk 3]
```diff
@@ -87,6 +81,14 @@ Versions can only be deleted by Project Administrators.
...
+
+The revision history, accessible by clicking the related icon inside the branch selection area, contains a list of snapshots of the Project configuration, created each time it has been saved, on a certain revision or version. It also includes useful information such as the snapshot date, author, associated message, and any tags created with it. Moreover, the right-side button will show the differences between the selected snapshot and the previous one, in order to have a better understanding of the changes applied to the revision.
```
Comment (GiovannaMonti) [inline docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:90]:
```suggestion
The revision history is accessible by clicking the related icon inside the branch selection area. It contains a list of snapshots of the Project configuration on the current revision or version, created each time it has been saved. It also includes useful information such as the snapshot date, author, associated message, and any tags associated with it. Moreover, the right-side button allows the user to visualize changes between the selected snapshot and the previous one, in order to better understand the configuration's evolution over time.
```

Comment (GiovannaMonti) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:90]: I would add a snapshot of the diff editor

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md (hunk 1 above)
Comment (ThisIsDemetrio) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:86]: Thanks for the catch. Probably a mistake after the merge from main.

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md (hunk 2 above)
Comment (ThisIsDemetrio) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:88]: Coming.

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md (hunk 3 above)
Comment (ThisIsDemetrio) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:90]: Done.

**Review by ThisIsDemetrio — APPROVED** [review]
LGTM!

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md [hunk 4]
```diff
@@ -87,6 +81,16 @@ Versions can only be deleted by Project Administrators.
 
 ![Versions management page](img/versions-overview.png)
 
+### Revision history
+
+To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. To keep track of how a revision has changed over time, the user can access its revision history, which contains past snapshots of the Project configuration.
```
Comment (fredmaggiowski) [inline docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:86]: We start the paragraph for revision history speaking about versions; that sounds odd.

I'd keep this paragraph in its previous location or rephrase it to emphasize revision history

```suggestion
To keep track of how a revision has changed over time, the user can access its revision history, which contains past snapshots of the Project configuration.
```

And put somewhere else the sentence 

```
To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. 
```

Comment (ThisIsDemetrio) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md]: Description of `version` has been moved to the Version paragraph describing the Versions Management page. Let me know if it looks better.

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md [hunk 5]
```diff
@@ -69,24 +69,28 @@ This change also means that the Console will not rely on the typical Git-based i
...
-![Revision History modal](img/revision-history.png)
-
-The table includes a list of all snapshots of the project configuration created with each save, along with useful information such as the snapshot date, author, associated message, and any tags created with it.
-
 Developers can manage *revisions* and *versions* from the dedicated sections inside the Overview area. Revisions can be accessed and deleted from the Revisions management page:
 
 ![Revisions management page](img/revisions-overview.png)
 
-Versions with their description and release note can be viewed and deleted through the versions management page:
+Versions, a feature similar to Git _tags_ that works as immutable past snapshots of the configuration in a particular moment, with their description and release note, can be viewed and deleted through the versions management page:
```
Comment (fredmaggiowski) [inline docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md:76]:
```suggestion
Versions works as immutable snapshots of the configuration in a particular moment, with their description and release note, can be viewed and deleted through the versions management page:
```

File: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md (hunk 4 above)
Comment (fredmaggiowski) [reply docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md]: sounds great. I've removed references to git tags.

**Review by fredmaggiowski — APPROVED** [review]
LGTM
